packages/block-editor/src/components/block-navigation/tree.js

```diff
@@ -3,29 +3,57 @@
  */
 
 import { __experimentalTreeGrid as TreeGrid } from '@wordpress/components';
-import { useEffect, useMemo, useRef } from '@wordpress/element';
+import { useDispatch } from '@wordpress/data';
+import { useCallback, useEffect, useMemo, useRef } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
+
 /**
  * Internal dependencies
  */
 import BlockNavigationBranch from './branch';
 import { BlockNavigationContext } from './context';
+import useBlockNavigationClientIds from './use-block-navigation-client-ids';
 import useBlockNavigationDropZone from './use-block-navigation-drop-zone';
+import { store as blockEditorStore } from '../../store';
 
 /**
  * Wrap `BlockNavigationRows` with `TreeGrid`. BlockNavigationRows is a
  * recursive component (it renders itself), so this ensures TreeGrid is only
  * present at the very top of the navigation grid.
  *
- * @param {Object}  props                                          Components props.
- * @param {boolean} props.__experimentalFeatures                   Flag to enable experimental features.
- * @param {boolean} props.__experimentalPersistentListViewFeatures Flag to enable features for the Persistent List View experiment.
+ * @param {Object}   props                                          Components props.
+ * @param {Array}    props.blocks                                   Custom subset of block client IDs to be used instead of the default hierarchy.
+ * @param {Function} props.onSelect                                 Block selection callback.
+ * @param {Array}    props.selectedBlockClientIds                   The client IDs of the (single or multi-) selected blocks.
+ * @param {boolean}  props.showNestedBlocks                         Flag to enable displaying nested blocks.
+ * @param {boolean}  props.showOnlyCurrentHierarchy                 Flag to limit the list to the current hierarchy of blocks.
+ * @param {boolean}  props.__experimentalFeatures                   Flag to enable experimental features.
+ * @param {boolean}  props.__experimentalPersistentListViewFeatures Flag to enable features for the Persistent List View experiment.
  */
 export default function BlockNavigationTree( {
+	blocks,
+	showOnlyCurrentHierarchy,
+	onSelect = () => {},
```

Comment:
I see we moved this and defined a `noop` in index, but it looks like we are still doing the same anonymous approach here.